sw/airborne/boards/bebop/mt9f002.c

```diff
@@ -781,6 +781,52 @@ static void mt9f002_set_gains(struct mt9f002_t *mt)
   write_reg(mt, MT9F002_GREEN2_GAIN, mt9f002_calc_gain(mt->gain_green2), 2);
 }
 
+void mt9f002_set_resolution(struct mt9f002_t *mt)
+{
+	/* Set output resolution */
+	  write_reg(mt, MT9F002_X_OUTPUT_SIZE, mt->output_width, 2);
+	  write_reg(mt, MT9F002_Y_OUTPUT_SIZE, mt->output_height, 2);
+	  /* Set scaling */
+	  uint16_t scaleFactor = ceil((float)MT9F002_SCALER_N / mt->output_scaler);
+	  mt->output_scaler = (float)MT9F002_SCALER_N / scaleFactor;
+	  int x_skip_factor = (mt->x_odd_inc + 1) / 2;
+	  int y_skip_factor = (mt->y_odd_inc + 1) / 2;
+	  mt->scaled_width 	= ceil((float)mt->output_width / mt->output_scaler) * x_skip_factor - mt->x_odd_inc;
+	  mt->scaled_height = ceil((float)mt->output_height / mt->output_scaler) * y_skip_factor - mt->y_odd_inc;
+	  if (mt->output_scaler != 1.0)
+	  {
+	    write_reg(mt, MT9F002_SCALING_MODE, 2, 2); // Vertical and horizontal scaling
+	    write_reg(mt, MT9F002_SCALE_M, scaleFactor, 2);
+	  }
+	  printf("OUTPUT_SIZE: (%i, %i)\tSCALED_SIZE: (%i, %i)\n", mt->output_width, mt->output_height, mt->scaled_width, mt->scaled_height);
```

Comment:
The other debugging printouts have a specific format so I assume you should follow that.